AWS_Backup/AMI_Backup.py

```diff
@@ -0,0 +1,245 @@
+    """
+    client = boto3.client('ec2', region_name=region)
+    ec2 = boto3.resource('ec2', region_name=region)
+    response2 = client.describe_images(
+        Filters=[
+            {
+                'Name': 'tag:DELETEON',
+                'Values': [
+                    'yes',
+                ]
+            },
+        ]
+    )
+    right_now_days_ago = datetime.datetime.today() - datetime.timedelta(days=days_del)
+    old_date = right_now_days_ago.replace(tzinfo=Utc)
+
+    for i in response2['Images']:
+        if i['CreationDate'] < str(old_date):
+            image_id = i['ImageId']
+            print image_id
+            delimage = ec2.Image(image_id)
+            snap_list = []
+            for j in i['BlockDeviceMappings']:
+                if 'Ebs'in j:
+                    snap_list.append(j['Ebs']['SnapshotId'])
+            try:
+
+                response = delimage.deregister()
+                for k in range(len(snap_list)):
+                    snapshot = ec2.Snapshot(snap_list[k])
+                    reponse = snapshot.delete()
+                    print "AMI snapshot ID deleted " + snap_list[k]
+            except Exception as e:
+                print e
+                message = 'Error while deleting image\nImageId:'+image_id+' \
+                                  \nRegion:'+region+'\nException:'+str(e)
+                if slack_req == 'true':
+                    slack(message, slack_channel, slack_webhook)
+                else:
+                    print message
+
+    response = client.describe_instances(
+        Filters=[
+            {
+                'Name': 'tag:AMIBACKUPON', #Tag used to identify list of Instances to be backed up.
+                'Values': [
+                    'yes',
+                ]
+            }
+        ]
+    )
+
+    for i in response['Reservations']:
+        for j in i['Instances']:
+            print j['InstanceId']
+            iid = j['InstanceId']
+            tag_key_list = []
+            tag_value_list = []
+            instance = ec2.Instance(iid)
+            for k in instance.tags:
+                tag_key_list.append(k['Key'])
+                tag_value_list.append(k['Value'])
+                if k['Key'] == 'Name':
+                    dscrip = k['Value']
+            try:
+                image = instance.create_image(
+                    Name=dscrip+ "-" +str(datetime.datetime.now().strftime('%Y-%m-%d_%H_%M_%S.%f')),
+                    NoReboot=True
+                    )
+                for l in range(len(tag_key_list)):
+                    tag = image.create_tags(
+                        Tags=[
+                            {
+                                'Key': tag_key_list[l],
+                                'Value': tag_value_list[l]
+                            }
+                        ]
+                    )
+
+                tag = image.create_tags(
+                    Tags=[
```

Comment:
move this to a smaller method. which would get the tags for deletion.

Reply:
Moved entire block to different module